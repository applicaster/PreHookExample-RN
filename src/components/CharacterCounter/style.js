import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  charactersRemainingContainer: {
    borderRadius: 10,
    flexDirection: 'column',
    height: 22,
    justifyContent: 'center',
    paddingHorizontal: 5,
  },
  charactersRemainingContainerWrapper: {
    backgroundColor: '#FFFFFF',
    borderRadius: 10,
    flexDirection: 'column',
    height: 22,
    justifyContent: 'center',
    marginLeft: 11,
  },
  charactersRemainingLabel: {
    fontSize: 10,
  },
});
