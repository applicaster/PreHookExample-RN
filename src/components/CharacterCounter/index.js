import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  View,
} from 'react-native';
import { styles } from './style';

class CharacterCounter extends Component {
  render() {
    const { backgroundColor = '#FFFFFF', textColor = '#000000' } = this.context;
    const backgroundColorStyle = { backgroundColor: `${textColor}` };
    let textColorStyle = { color: backgroundColor };

    let { currentCharacters, maxCharacters } = this.props;
    if (currentCharacters > 999) {
      currentCharacters = '+999';
    }

    if (currentCharacters > maxCharacters) {
      textColorStyle = { color: '#FF0000' };
    }

    const charactersRemainingLabel = `${currentCharacters}/${maxCharacters}`;

    return (
      <View style={[styles.charactersRemainingContainer, backgroundColorStyle]}>
        <Text style={textColorStyle}>{charactersRemainingLabel}</Text>
      </View>
    );
  }
}

CharacterCounter.propTypes = {
  currentCharacters: PropTypes.number,
  maxCharacters: PropTypes.number,
};

CharacterCounter.contextTypes = {
  backgroundColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default CharacterCounter;
