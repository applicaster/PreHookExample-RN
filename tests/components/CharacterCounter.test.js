import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CharacterCounter from '../../src/components/CharacterCounter';

describe('CharacterCounter', () => {
  test('component renders correctly when number of characters is within limit', () => {
    const html = renderer.create(
      <CharacterCounter maxCharacters={10} currentCharacters={1} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when number of characters is over limit', () => {
    const html = renderer.create(
      <CharacterCounter maxCharacters={10} currentCharacters={11} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when number of characters is over 3 digits', () => {
    const html = renderer.create(
      <CharacterCounter maxCharacters={10} currentCharacters={1000} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
