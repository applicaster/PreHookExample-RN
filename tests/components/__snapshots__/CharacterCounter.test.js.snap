// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCounter component renders correctly when number of characters is over 3 digits 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "flexDirection": "column",
        "height": 22,
        "justifyContent": "center",
        "marginLeft": 11,
        "paddingHorizontal": 5,
      },
      Object {
        "backgroundColor": "#000000",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FF0000",
      }
    }
  >
    +999/10
  </Text>
</View>
`;

exports[`CharacterCounter component renders correctly when number of characters is over limit 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "flexDirection": "column",
        "height": 22,
        "justifyContent": "center",
        "marginLeft": 11,
        "paddingHorizontal": 5,
      },
      Object {
        "backgroundColor": "#000000",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FF0000",
      }
    }
  >
    11/10
  </Text>
</View>
`;

exports[`CharacterCounter component renders correctly when number of characters is within limit 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "flexDirection": "column",
        "height": 22,
        "justifyContent": "center",
        "marginLeft": 11,
        "paddingHorizontal": 5,
      },
      Object {
        "backgroundColor": "#000000",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
  >
    1/10
  </Text>
</View>
`;
